pfinance/functions.py

```diff
@@ -63,6 +63,38 @@ def future_value_series(
     return payment * start_modifier * (((1 + effective_rate) ** total_periods) - 1) / effective_rate
 
 
+def present_value(
+    payment: float,
+    interest_rate: float,
+    periods: int,
+    future_value: float = 0,
+    start_of_period: bool = False
+) -> float:
+    '''
+    Returns the present value of a loan or investment based on a constant interest rate.
+
+        Parameters:
+            payment (float): The regular payment made each period
+            interest_rate (float): The interest rate per period, e.g. year
+            periods (int): Number of payments made over the term of the investment
+            future_value (float): The total cash amount you want to have at the last payment, default 0
+            start_of_period (bool): Payment is made at start of each period, default False
+
+        Returns:
+            present_value (float): Present value of a loan or investment
+    '''
+    if interest_rate == 0:
+        return (-1 * payment * periods) - future_value
+
+    if start_of_period:
+        pv_type = 1
+    else:
+        pv_type = 0
+
+    return -1 * (payment * (1 + interest_rate * pv_type) * ((1 + interest_rate) ** periods - 1)
+                 / interest_rate + future_value) / ((1 + interest_rate) ** periods)
```

Comment:
```suggestion
    start_modifier = 1
    if start_of_period:
        start_modifier += interest_rate

    return -1 * (payment * start_modifier * ((1 + interest_rate) ** periods - 1) / interest_rate + future_value) / ((1 + interest_rate) ** periods)
```
line length on the return may still be too long, but with this you could probably reformat it so that just the final denominator is on its own line

Reply:
I think breaking up the formula into a numerator and denominator will make it easier to read.